xbmc/dialogs/GUIDialogSelectColor.h

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (C) 2005-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "GUIDialogBoxBase.h"
+#include "view/GUIViewControl.h"
+
+#include <string>
+
+class CFileItem;
+class CFileItemList;
+
+class CGUIDialogSelectColor : public CGUIDialogBoxBase
+{
+public:
+  CGUIDialogSelectColor(void);
+  ~CGUIDialogSelectColor(void) override;
+  bool OnMessage(CGUIMessage& message) override;
+  bool OnBack(int actionID) override;
+  void Reset();
+  std::string GetSelectedColor() const;
+  void SetSelectedColor(const std::string& hexColor);
+  void SetButtonFocus(bool buttonFocus);
+
+  /*! \brief Set a XML file path where to load colors instead to load the default colors
+    \param filePath The XML file path
+  */
+  void SetFilePath(std::string& filePath) { m_filePath = filePath; }
+
+protected:
+  CGUIControl* GetFirstFocusableControl(int id) override;
+  void OnWindowLoaded() override;
+  void OnInitWindow() override;
+  void OnDeinitWindow(int nextWindowID) override;
+  void OnWindowUnload() override;
+
+  virtual void OnSelect(int idx);
+
+private:
+  int GetSelectedItem() const;
+
+  CGUIViewControl m_viewControl;
+  CFileItemList* m_vecList;
+  bool m_focusToButton{};
```

Comment:
was the intention to initialize the bool member? you don't default-init the other members...

Reply:
changed